### PR TITLE
Fix 476 combine array entries

### PR DESCRIPTION
**Pull request by damhuonglan**
### What does this PR do?
SmartSolo split data for the same deployment into different files. As the result there are different entries for the same deployment in array table. 
This PR modify segdtoph5 to combine array entries within the same deployment (with overlap or no gap) into one entry. This combination is applied for SmartSolo only. 
### Checklist
- [ ] This PR is not directly related to an existing issue (which has no PR yet).
- [x] All tests pass.
- [ ] Any new or changed features have are documented.
- [ ] Changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

**Comment by ascire-pic** [comment]
Looks good; tested with a few different subsets of data including continuous SmartSolo data, SmartSolo data with a gap, and Fairfield data. All resulting array tables appear to be correct.

**Comment by damhuonglan** [comment]
@ascire-pic I'm sorry that I assigned you this PR instead of asking you to review it. Can you go to the top of the PR to approve it please?

**Review by ascire-pic — APPROVED** [review]
Copying previous comment re testing:
Looks good; tested with a few different subsets of data including continuous SmartSolo data, SmartSolo data with a gap, and Fairfield data. All resulting array tables appear to be correct.

**Review by timronan — COMMENTED** [review]
Are the DAS tables combined or only the array tables? Where is the variable aOfDas defined? I see that it is an input for combine_array_entries. Is `aOfDas` the DAS list that is returned by `query_das_t`?

What does "the times are combined if gap less than 2m" this mean? I don't see a time cutoff logic in the code but may be missing something. 

Overall, I think this code is ready to merge but just want to make sure that I understand the documentation. Thanks for building this.

File: ph5/utilities/segd2ph5.py [hunk 1]
```diff
@@ -990,6 +992,63 @@ def write_arrays(Array_t):
...
+            deployTime = d['deploy_time/epoch_l']
+            if deployTime > prevPickupTime:
+                currDeployTime = deployTime
+                if deployTime not in aOnDeployTimes:
+                    aOnDeployTimes[deployTime] = {}
+                if c not in aOnDeployTimes[deployTime]:
+                    aOnDeployTimes[deployTime][c] = [d]
+            else:
+                uEntry = aOnDeployTimes[currDeployTime][c][0]
+                msg = "Das %s - %s - station %s - chan %s: " % (
```
[the remark this answers is not in the document]
Comment (damhuonglan) [reply ph5/utilities/segd2ph5.py:1030]: @timronan If you look in the function write_arrays() at line 980,881, you can see that Array_t[line][das] is feed to combine_array_entries(). This is a dictionary for all array entries of a das, each entry if of a deploy time. This function will create a new dictionary in which all entries that it decides to be in the same deployment  are combined into one and return this dictionary to replace the one that was passed to the function. The write_arrays will use the data returned to create entries in array table. 

segdtoph5 is the tool to create ph5 data from raw data, we can't use query_das_t when the data aren't formed yet.

The comment "the times are combined if gap less than 2m" should be removed. At the beginning I coded in that way, but now change to the entries with overlaps or no gaps will be combined into one entry.

[the remark this answers is not in the document]
Comment (timronan) [reply ph5/utilities/segd2ph5.py:1030]: @damhuonglan thanks for the explanation. This PR seems ready to merge.

**Pull request merged**